src/scripts/main.js

```diff
@@ -1 +1,17 @@
 'use strict';
+const calendar = document.querySelector('.calendar');
+
+const daysInMonth = 31;
+
+function generateCalendar(daysInMonth) {
+
+  // Add divs for each day in the month
+  for (let i = 1; i <= daysInMonth; i++) {
+    const day = document.createElement('div');
+    day.classList.add('calendar__day');
+    day.setAttribute('data-day', i);
+    calendar.appendChild(day);
+  }
+}
```

Comment:
Your function `generateCalendar` is directly manipulating the DOM inside the loop. Each change to the DOM can cause the page to re-render, which is a costly operation. It's better to create a document fragment, append your days to it inside the loop, and then append this fragment to the calendar. This way, you will only touch the DOM once, improving the performance of your code.